### PR TITLE
forc run output is formatted such that key! types and data fields are…

### LINKED ISSUE #1260: Make visualizing forc log output pretty
- Right now log data is pretty visually hard to parse with Forc.
- It would be nice to simplify this or have a simplified view or option available

For this program (as an exampe):
```
script;

fn log_b256(value: b256) {
    asm(r1: value, r2: 32) {
        logd zero zero r1 r2;
    }
}

fn main() {
    let k:b256 = 0xef86afa9696cf0dc6385e2c407a6e159a1103cefb7e2ae0636fb33d3cb2a9e4a;

    log_b256(k);
}
```

The `forc run` return looks like this:
```bash
[LogData { id: 0000000000000000000000000000000000000000000000000000000000000000, ra: 0, rb: 0, ptr: 10472, len: 32, digest: be240c408b034a8f3e67813d85c05d5379e02c17d4012440a4bf73450aa8935d, data: [239, 134, 175, 169, 105, 108, 240, 220, 99, 133, 226, 196, 7, 166, 225, 89, 161, 16, 60, 239, 183, 226, 174, 6, 54, 251, 51, 211, 203, 42, 158, 74], pc: 10416, is: 10352 }, Return { id: 0000000000000000000000000000000000000000000000000000000000000000, val: 0, pc: 10420, is: 10352 }, ScriptResult { result: InstructionResult { reason: RESERV00, instruction: Instruction { op: 0, ra: 0, rb: 0, rc: 0, rd: 0, imm06: 0, imm12: 0, imm18: 0, imm24: 0 } }, gas_used: 424 }]
```

Could look like this:
```bash
Logs Returned by Execution:

LogData:
    Register A: 0
    Register B: 0
    Data: 0xef86afa9696cf0dc6385e2c407a6e159a1103cefb7e2ae0636fb33d3cb2a9e4a
```

Having some kind of easy to parse clean version would be nice. If I want all the other program details I could just maybe flag an option for it.

### PR DESCRIPTION
**Pull request by kayagokalp**
closes #1260.

Output for the example program posted in the issue:

- `forc run --pretty-print`

```
[
  {
    "LogData": {
      "data": "ef86afa9696cf0dc6385e2c407a6e159a1103cefb7e2ae0636fb33d3cb2a9e4a",
      "digest": "be240c408b034a8f3e67813d85c05d5379e02c17d4012440a4bf73450aa8935d",
      "id": "0000000000000000000000000000000000000000000000000000000000000000",
      "is": 10352,
      "len": 32,
      "pc": 10416,
      "ptr": 10488,
      "ra": 0,
      "rb": 0
    }
  },
  {
    "Return": {
      "id": "0000000000000000000000000000000000000000000000000000000000000000",
      "is": 10352,
      "pc": 10424,
      "val": 0
    }
  },
  {
    "ScriptResult": {
      "gas_used": 435,
      "result": "Success"
    }
  }
]
```

- `forc run` 
```
[{"LogData":{"data":"ef86afa9696cf0dc6385e2c407a6e159a1103cefb7e2ae0636fb33d3cb2a9e4a","digest":"be240c408b034a8f3e67813d85c05d5379e02c17d4012440a4bf73450aa8935d","id":"0000000000000000000000000000000000000000000000000000000000000000","is":10352,"len":32,"pc":10416,"ptr":10488,"ra":0,"rb":0}},{"Return":{"id":"0000000000000000000000000000000000000000000000000000000000000000","is":10352,"pc":10424,"val":0}},{"ScriptResult":{"gas_used":435,"result":"Success"}}]
```

### CODE REVIEW

**Review by adlerjohn — COMMENTED** [review]
CI failing

**Review by mohammadfawaz — DISMISSED** [review]
Looks good!

File: forc/Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,7 @@ fuel-vm = "0.11"
 futures = "0.3"
 hex = "0.4.3"
 serde = { version = "1.0", features = ["derive"] }
-serde_json = "1.0.73"
+serde_json = { version = "1.0.73", features = ["preserve_order"] }
```
Comment (adlerjohn) [inline forc/Cargo.toml:34]: Is this necessary for this PR? If not, could you remove and do this in a separate PR?

**Review by adlerjohn — CHANGES_REQUESTED** [review]

File: forc/Cargo.toml (hunk 1 above)
Comment (kayagokalp) [reply forc/Cargo.toml:34]: Since I am converting `Vec<u8>` fields to hex strings `serde_json` changes the order the receipt is printed. (data is printed at the beginning if no `preserve_order` is used) if that is not a problem I can omit this one or do that in separate PR.

Comment (adlerjohn) [reply forc/Cargo.toml:34]: It's not a bad change to please, but please do it in a separate PR if it's not needed to resolve pretty-printing receipts.

Comment (adlerjohn) [reply forc/Cargo.toml:34]: Ah I may have misunderstood. The issue right now is that `features = ["preserve_order"]` seems to affect the JSON ABI files (see all the changed oracle files). That's what shouldn't be changed as part of this PR is what I meant.

**Comment by mohammadfawaz** [comment]
I just noticed that there are `"` added to the receipts everywhere now (e.g. `"LogData"` instead of `LogData`). Can we remove those?

File: forc/Cargo.toml (hunk 1 above)
Comment (kayagokalp) [reply forc/Cargo.toml:34]: Hmm, without changing JSON ABIs I wouldn't be able to use `features = ["preserve_order"]` so I think it makes perfect sense to do that addition in separate PR since it is used to fix ordering not pretty-printing JSON (with hex strings)

Comment (mohammadfawaz) [reply forc/Cargo.toml:34]: Is the order in which the receipts show up not important?

**Comment by kayagokalp** [comment]
> I just noticed that there are " added to the receipts everywhere now (e.g. "LogData" instead of LogData). Can we remove those?

The receipt is printed as JSON and serialized by `serde_json` so the `"` are coming from there. I guess If we remove`"` output wouldn't be a valid JSON anymore. But if that is something we would be ok with we may need to add a custom serialize implementation to the Receipt directly (or if there are any other suggestions that would be fantastic).

**Comment by adlerjohn** [comment]
The receipts printed out should be valid JSON. I think that does require the `""`.

**Comment by kayagokalp** [comment]
Currently, the output is 3 separate valid JSONs (since there are 3 Receipts printed):

```
{
  "LogData": {
    "id": "0000000000000000000000000000000000000000000000000000000000000000",
    "ra": 0,
    "rb": 0,
    "ptr": 10488,
    "len": 32,
    "digest": "be240c408b034a8f3e67813d85c05d5379e02c17d4012440a4bf73450aa8935d",
    "data": "ef86afa9696cf0dc6385e2c407a6e159a1103cefb7e2ae0636fb33d3cb2a9e4a",
    "pc": 10416,
    "is": 10352
  }
}
```

```
{
  "Return": {
    "id": "0000000000000000000000000000000000000000000000000000000000000000",
    "val": 0,
    "pc": 10424,
    "is": 10352
  }
}
```

```
{
  "ScriptResult": {
    "result": "Success",
    "gas_used": 435
  }
}
```

But I can also make the output like the following:

```
[{
	"LogData": {
		"id": "0000000000000000000000000000000000000000000000000000000000000000",
		"ra": 0,
		"rb": 0,
		"ptr": 10488,
		"len": 32,
		"digest": "be240c408b034a8f3e67813d85c05d5379e02c17d4012440a4bf73450aa8935d",
		"data": "ef86afa9696cf0dc6385e2c407a6e159a1103cefb7e2ae0636fb33d3cb2a9e4a",
		"pc": 10416,
		"is": 10352
	}
}, {
	"Return": {
		"id": "0000000000000000000000000000000000000000000000000000000000000000",
		"val": 0,
		"pc": 10424,
		"is": 10352
	}
}, {
	"ScriptResult": {
		"result": "Success",
		"gas_used": 435
	}
}]
```

which is a single valid JSON

File: forc/Cargo.toml (hunk 1 above)
Comment (kayagokalp) [reply forc/Cargo.toml:34]: I am not aware if there would be any technical issues arising from the changed order, I did this to match the current implementation (and other receipt types).

Comment (adlerjohn) [reply forc/Cargo.toml:34]: The order of _receipts_ is important since receipts are ordered. The order of _fields_ within the receipts isn't important as they all have a key.

**Comment by kayagokalp** [comment]
After the latest changes (which reverts the usage of `preserve_order` and resulting abi changes) output looks like the following (also updating the PR explanation with this one):

`forc run --pretty-print`

```
[
  {
    "LogData": {
      "data": "ef86afa9696cf0dc6385e2c407a6e159a1103cefb7e2ae0636fb33d3cb2a9e4a",
      "digest": "be240c408b034a8f3e67813d85c05d5379e02c17d4012440a4bf73450aa8935d",
      "id": "0000000000000000000000000000000000000000000000000000000000000000",
      "is": 10352,
      "len": 32,
      "pc": 10416,
      "ptr": 10488,
      "ra": 0,
      "rb": 0
    }
  },
  {
    "Return": {
      "id": "0000000000000000000000000000000000000000000000000000000000000000",
      "is": 10352,
      "pc": 10424,
      "val": 0
    }
  },
  {
    "ScriptResult": {
      "gas_used": 435,
      "result": "Success"
    }
  }
]
```

`forc run`

```
[{"LogData":{"data":"ef86afa9696cf0dc6385e2c407a6e159a1103cefb7e2ae0636fb33d3cb2a9e4a","digest":"be240c408b034a8f3e67813d85c05d5379e02c17d4012440a4bf73450aa8935d","id":"0000000000000000000000000000000000000000000000000000000000000000","is":10352,"len":32,"pc":10416,"ptr":10488,"ra":0,"rb":0}},{"Return":{"id":"0000000000000000000000000000000000000000000000000000000000000000","is":10352,"pc":10424,"val":0}},{"ScriptResult":{"gas_used":435,"result":"Success"}}]
```

**Review by adlerjohn — APPROVED** [review]
